app/src/main/java/com/babylon/wallet/android/presentation/rootdetection/RootDetectionDialog.kt

```diff
@@ -0,0 +1,64 @@
+package com.babylon.wallet.android.presentation.rootdetection
+
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import com.babylon.wallet.android.MainActivity
+import com.babylon.wallet.android.R
+import com.babylon.wallet.android.designsystem.theme.RadixTheme
+import com.babylon.wallet.android.presentation.ui.composables.BasicPromptAlertDialog
+
+const val ROUTE_ROOT_DETECTION = "root_detection_route"
+
+@Composable
+fun RootDetectionContent(
+    viewModel: RootDetectionViewModel,
+    onAcknowledgeDeviceRooted: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val activity = (LocalContext.current as MainActivity)
+    BackHandler {
+        activity.finish()
+    }
+    LaunchedEffect(Unit) {
+        viewModel.oneOffEvent.collect {
+            when (it) {
+                RootDetectionEvent.RootAcknowledged -> onAcknowledgeDeviceRooted()
+            }
+        }
+    }
+    Box(
+        modifier = modifier
+            .fillMaxSize()
+            .background(RadixTheme.colors.blue1)
+            .padding(horizontal = RadixTheme.dimensions.paddingDefault)
+    ) {
+        BasicPromptAlertDialog(
+            finish = {
+                if (it) {
+                    viewModel.onAcknowledgeClick()
+                } else {
+                    activity.finish()
+                }
+            },
+            title = {
+                Text(
+                    text = "Hey, this device looks rooted!  Don’t use the Wallet!", // TODO Crowdin
+                    style = RadixTheme.typography.body2Header,
```

Comment:
I see that in most places we use body1Header for a title in prompt dialogs. Is this intentional? Maybe we want to be consistent? If so, there is signature of `BasicPromptAlertDialog` that expects just a string for title and uses default style

Reply:
> I see that in most places we use body1Header for a title in prompt dialogs. Is this intentional? Maybe we want to be consistent? If so, there is signature of `BasicPromptAlertDialog` that expects just a string for title and uses default style

Hmm `NotSecureDialog` and `IncompatibleProfileContent` use **body2Header** in turn so Im not really sure. But yeah I will align it for this case.